core/templates/components/state-editor/state-skill-editor/state-skill-editor.directive.ts

```diff
@@ -0,0 +1,135 @@
+                untriagedSkillSummaries = response.untriagedSkillSummaries;
+                $rootScope.$applyAsync();
+              });
+          };
+
+          $scope.addSkillModalAsync = async function() {
+            var sortedSkillSummaries = (
+              StoryEditorStateService.getSkillSummaries());
+            var allowSkillsFromOtherTopics = true;
+            var skillsInSameTopicCount = 0;
+            $uibModal.open({
+              templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                '/components/skill-selector/select-skill-modal.template.html'),
+              backdrop: 'static',
+              resolve: {
+                skillsInSameTopicCount: () => skillsInSameTopicCount,
+                sortedSkillSummaries: () => sortedSkillSummaries,
+                categorizedSkills: () => categorizedSkills,
+                allowSkillsFromOtherTopics: () => allowSkillsFromOtherTopics,
+                untriagedSkillSummaries: () => untriagedSkillSummaries
+              },
+              controller: 'SelectSkillModalController',
+              windowClass: 'skill-select-modal',
+              size: 'xl'
+            }).result.then(function(result) {
+              try {
+                StateSkillService.displayed = result.id;
+                StateSkillService.saveDisplayedValue();
+                $scope.onSaveLinkedSkillId(result.id);
+              } catch (err) {
+                AlertsService.addInfoMessage(
+                  err, 5000);
+              }
+            }, function() {
+              // Note to developers:
+              // This callback is triggered when the Cancel button is clicked.
+              // No further action is needed.
+            });
+          };
+
+          $scope.deleteSkill = function() {
+            AlertsService.clearWarnings();
+            $uibModal.open({
+              templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                '/pages/exploration-editor-page/editor-tab/templates/' +
+                'modal-templates/delete-state-skill-modal.template.html'),
+              backdrop: true,
+              controller: 'ConfirmOrCancelModalController'
+            }).result.then(function() {
+              StateSkillService.displayed = null;
+              StateSkillService.saveDisplayedValue();
+              $scope.onSaveLinkedSkillId(StateSkillService.displayed);
+            }, function() {
+              // Note to developers:
+              // This callback is triggered when the Cancel button is clicked.
+              // No further action is needed.
+            });
+          };
+
+          $scope.getSkillEditorUrl = function() {
+            if (StateSkillService.displayed) {
+              return '/skill_editor/' + StateSkillService.displayed;
+            }
+          };
+
+          $scope.toggleSkillEditor = function() {
+            $scope.skillCardIsShown = !$scope.skillCardIsShown;
```

Comment:
Should it be "skillEditorCardIsShown" or "skillEditorIsShown", given the name of this function?

Reply:
Done.